### PR TITLE
ENG-19307: Snapshots do not work in memcheck mode

### PR DESCRIPTION
**Pull request by n3world**

### CODE REVIEW

File: tests/frontend/org/voltdb/utils/TestSnapshotComparer.java [hunk 1]
```diff
@@ -114,10 +116,11 @@ public void testSimpleFaiure() {
     // positive test
     // Cluster with no rejoin/snapshot should always has same row order
     @Test
-    public void testSimpleSuccess() throws IOException, ProcCallException {
-        if (!MiscUtils.isPro()) {
+    public void testSimpleSuccess() throws Exception {
+        if (LocalCluster.isMemcheckDefined()) {
             return;
```
Comment (rdykiel) [inline tests/frontend/org/voltdb/utils/TestSnapshotComparer.java:121]: why do we bypass the test in this case?

**Comment by zeemanhe** [comment]
Is SnapshotComparer or SnapshotSave not working with  -Dbuild=memcheck ?

File: tests/frontend/org/voltdb/utils/TestSnapshotComparer.java (hunk 1 above)
Comment (n3world) [reply tests/frontend/org/voltdb/utils/TestSnapshotComparer.java:121]: Snapshots do not work in memcheck mode

**Comment by n3world** [comment]
> Is SnapshotComparer or SnapshotSave not working with -Dbuild=memcheck ?

No snapshots have never worked with memcheck builds because snapshots were not implemented for the ipc ee. That is why so many tests use the `overrideAnyRequestForValgrind()`

**Pull request merged**